packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -156,6 +158,23 @@ function hasEcsCredentials(): boolean {
   return (AWS.ECSCredentials.prototype as any).isConfiguredForEcsCredentials();
 }
 
+/**
+ * Return whether it looks like we'll have EKS credentials available
+ * No check like hasEcsCredentials available, so have to implement our own.
+ * @see https://github.com/aws/aws-sdk-js/blob/3ccfd94da07234ae87037f55c138392f38b6881d/lib/credentials/token_file_web_identity_credentials.js#L59
+ */
+function hasEksCredentials(): boolean {
+  // eslint-disable-next-line no-console
+  console.log('HasEksCreds has been called');
+  // eslint-disable-next-line no-console
+  console.log(`EksCredsState: ${Boolean(process && process.env && (process.env.AWS_ROLE_ARN && process.env.AWS_WEB_IDENTITY_TOKEN_FILE))}`);
+  // eslint-disable-next-line no-console
+  console.log(`EksEnv.AWS_ROLE_ARN: ${process.env.AWS_ROLE_ARN}`);
+  // eslint-disable-next-line no-console
+  console.log(`EksEnv.AWS_WEB_IDENTITY_TOKEN_FILE: ${process.env.AWS_WEB_IDENTITY_TOKEN_FILE}`);
+  return Boolean(process && process.env && (process.env.AWS_ROLE_ARN && process.env.AWS_WEB_IDENTITY_TOKEN_FILE));
```

Comment:
Why not `return process.env.AWS_ROLE_ARN && process.env.AWS_WEB_IDENTITY_TOKEN_FILE`

Reply:
What @hoegertn said :)